### ISSUE TITLE
[FEATURE] Making 'global shared vision' turn on when party are close together, and turn off if they split up too much

### ISSUE THREAD

**Issue by nirurin** [CDeenen/SharedVision#30]
The idea being - if the party are playing as a group (say, within talking distance) then their vision should be shared, as they are passing information around and they can all hear when a creature pops up nearby even if they can't see it etc. Means that noone misses any of the action if they're near each other. 

But then if they split up too much, they loose that benefit, and return to standard 'unshared' vision.

**Comment by CDeenen** [comment]
I like this feature, I'll see what I can do.

**Comment by nirurin** [comment]
That would be awesome Thankyou. I'm aware it's probably possible using a
world macro but that would be so far beyond me that I can't event begin to
attempt it haha

**Comment by CDeenen** [comment]
I'm thinking of approaching it slightly different from how you might imagine (based on what you've said, but I might've misinterpreted you). It'd be independent of the 'global shared vision' setting. 
There'd be a configurable radius in the module settings, let's say 30ft. Any tokens within 30ft of your selected or owned tokens will share their vision with you. This means that if your token is on the outskirts of the party, only part of the party might share their vision. This is a bit more flexible, since you might have a split party, each only seeing what the people in their direct surroundings can seen.
Let me know if you agree, otherwise I could consider using 'global shared vision'.

**Comment by nirurin** [comment]
No, what you have described is how I would have ideally wanted it to work!
Sounds great!

**Comment by nirurin** [comment]
Just following up, curious if you had any success with this?

**Comment by Weskernation** [comment]
> I'm thinking of approaching it slightly different from how you might imagine (based on what you've said, but I might've misinterpreted you). It'd be independent of the 'global shared vision' setting. There'd be a configurable radius in the module settings, **let's say 30ft. Any tokens within 30ft** of your selected or owned tokens will share their vision with you. This means that if your token is on the outskirts of the party, only part of the party might share their vision. This is a bit more flexible, since you might have a split party, each only seeing what the people in their direct surroundings can seen. Let me know if you agree, otherwise I could consider using 'global shared vision'.

Can this range be blocked by walls. It would be weird if the group is separated (by walls) but still withing this range making them share their vision. Perhaps also make the sharing disabled when the blinded condition is involved or other vision imparing effects
